### ISSUE TITLE
[member-ordering] Index signature is considered a method in interface

### ISSUE THREAD

**Issue by sveyret** [typescript-eslint/typescript-eslint#1186]
**Repro**

```JSON
{
  "rules": {
   "'@typescript-eslint/member-ordering": ["error"]
  }
}
```

```TS
interface Extra {
  [key: string]: any
  req: Request
  res: Response
}
```

**Expected Result**

No error.

**Actual Result**

Member req should be declared before all method definitions. eslint(@typescript-eslint/member-ordering)
Member res should be declared before all method definitions. eslint(@typescript-eslint/member-ordering)

**Additional Info**

If I remove the index signature, or put it after the interface members, there is no error.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.6.1` |
| `@typescript-eslint/parser`        | `2.6.1` |
| `TypeScript`                       | `3.6.4` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `12.13.0` |
| `npm`                              | `6.12.0` |

**Comment by bradzacher** [comment]
Index signature support was never added to the rule.

**Comment by sveyret** [comment]
Even without adding an option to support position of index signature, the rule should be corrected to, at least, ignore index signature when checking order. The tslint replaced rule was not failing in this situation.

**Comment by bradzacher** [comment]
> The tslint replaced rule was not failing in this situation.

We don't aim for 1:1 replacements with tslint when adding a rule. TSLint rules in many cases have options added that aren't used, or have options that should be on by default, etc.
When adding a rule that is analogous to one in TSLint, it's built for the use case.

The rule was built by a community member (as has been pretty much every rule).
At the time, they didn't have a use case for sorting index signatures, so they didn't add it.
None of the reviewers had an active use case for it either, so it was merged without it.

Either nobody has had a use case for this since the rule was implemented, or users have worked around it.

Happy to accept a PR to add support, if this functionality is important to you.

https://github.com/typescript-eslint/typescript-eslint/blob/c8fe51575d743ad317a09c18658c79d45059412b/packages/eslint-plugin/src/rules/member-ordering.ts#L197

**Comment by sveyret** [comment]
I'll try to see if I have time to create a PR. Unfortunately, I'm not sure I'll have time.

Just to be clear again: I'm not expecting the rule to allow to sort the index signature, but to at least ignore it.

**Comment by armano2** [comment]
fixed in #1190
